server/src/main/java/org/opensearch/index/mapper/IpFieldMapper.java

```diff
@@ -285,34 +288,44 @@ public Query termsQuery(List<?> values, QueryShardContext context) {
                 }
                 addresses[i++] = address;
             }
-            return InetAddressPoint.newSetQuery(name(), addresses);
+            Supplier<Query> dvQuery = () -> {
```

Comment:
The previous code has a bit of duplication, I agree, but adding supplier is not really helpful here (in my opinion), plus it generates "garbage" instances that may never be used.